exporter/src/main/as/flump/export/AtlasImpl.as

```diff
@@ -0,0 +1,141 @@
+                const tex :SwfTexture = node.texture;
+                const bm :Bitmap = new Bitmap(node.texture.toBitmapData(_borderSize), "auto", true);
+                constructed.addChild(bm);
+                bm.x = node.paddedBounds.x;
+                bm.y = node.paddedBounds.y;
+                collapsedBounds = collapsedBounds.union(node.paddedBounds);
+            });
+            _bitmapData = Util.renderToBitmapData(constructed,
+                    Util.nextPowerOfTwo(collapsedBounds.x + collapsedBounds.width),
+                    Util.nextPowerOfTwo(collapsedBounds.y + collapsedBounds.height),
+                    quailty);
+        }
+        return _bitmapData;
+    }
+
+    // Try to place a texture in this atlas
+    public function place (tex :SwfTexture, xx :uint, yy :uint) :void {
+        var w :int = tex.w + (_borderSize * 2);
+        var h :int = tex.h + (_borderSize * 2);
+        if (w > _width || h > _height) {
+            throw new Error("Tried to place a texture outside of the atlas. This is a bug.");
+        }
+        if (isMasked(xx, yy, w, h)) {
+            throw new Error("Tried to place a texture over another texture. This is a bug.");
+        }
+        var node :Node = new Node(xx, yy, _borderSize, tex);
+        _nodes.push(node);
+        setMasked(node.paddedBounds.x, node.paddedBounds.y, node.paddedBounds.width, node.paddedBounds.height);
+    }
+
+    protected static var _isMaskedPoint:Point = new Point();
+    protected static var _isMaskedRect:Rectangle = new Rectangle();
+    protected function isMasked (x :int, y :int, w :int, h :int) :Boolean {
+        _isMaskedRect.setTo(x, y, w, h);
+        return _mask.hitTest(_isMaskedPoint, 1, _isMaskedRect);
+    }
+
+    protected static var _setMaskedRect:Rectangle = new Rectangle();
+    protected function setMasked (x :int, y :int, w: int, h :int) :void {
+        _setMaskedRect.setTo(x, y, w, h);
+        _mask.fillRect(_setMaskedRect, 0xffffffff);
+    }
+
+    protected function maskAt (xx :int, yy :int) :Boolean {
+        return _mask.getPixel32(xx, yy) != 0;
+    }
+
+    protected var _nodes :Array = [];
+    protected var _width :int;
+    protected var _height :int;
+    protected var _borderSize :int;
+    protected var _mask :BitmapData;
+    protected var _bitmapData :BitmapData;
+    protected var _scaleFactor :int;
+    protected var _quality :String;
+}
+}
+
+import flash.geom.Rectangle;
+
+import flump.SwfTexture;
+
+class Node
+{
+    public var bounds :Rectangle;
+    public var paddedBounds :Rectangle;
+    public var texture :SwfTexture;
+
+    public function Node (x :int, y :int, borderSize :int, texture :SwfTexture) {
+        this.texture = texture;
+        this.bounds = new Rectangle(x + borderSize, y + borderSize, texture.w, texture.h);
+        this.paddedBounds = new Rectangle(x, y,
+                texture.w + (borderSize * 2),
+                texture.h + (borderSize * 2));
+    }
+}
+
```

Comment:
This class was doing duplicate work by trying to place things again in the texture. I removed it, now it only acts as a validator.